packages/components/src/components/select/select/select.tsx

```diff
@@ -500,10 +525,13 @@ export class Select {
     this.selectOptgroups = [];
 
     for (const child of Array.from(this.host.children).filter(
-      (el) => el.tagName !== 'SELECT' && el.slot !== 'label' && el.slot !== 'description' && el.slot !== 'message'
+      (el) =>
+        el.tagName !== 'SELECT' &&
+        el.slot !== 'label' &&
+        el.slot !== 'description' &&
+        el.slot !== 'message' &&
+        el.slot !== 'filter'
     )) {
-      throwIfElementIsNotOfKind(this.host, child as HTMLElement, ['p-select-option', 'p-optgroup']);
```

Comment:
This is still valid for when the `fliter` slot is unused. So maybe use it conditionally instead ?